src/main/java/org/terasology/additionalitempipes/action/UnificatorAction.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright 2017 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.additionalitempipes.action;
+
+import org.terasology.additionalitempipes.components.UnificatorComponent;
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.entitySystem.event.EventPriority;
+import org.terasology.entitySystem.event.ReceiveEvent;
+import org.terasology.entitySystem.prefab.Prefab;
+import org.terasology.entitySystem.systems.BaseComponentSystem;
+import org.terasology.entitySystem.systems.RegisterMode;
+import org.terasology.entitySystem.systems.RegisterSystem;
+import org.terasology.itempipes.controllers.PipeSystem;
+import org.terasology.itempipes.event.PipeInsertEvent;
+import org.terasology.logic.common.ActivateEvent;
+import org.terasology.logic.common.lifespan.LifespanComponent;
+import org.terasology.logic.inventory.PickupComponent;
+import org.terasology.math.Side;
+import org.terasology.math.geom.Vector3i;
+import org.terasology.physics.components.RigidBodyComponent;
+import org.terasology.registry.In;
+import org.terasology.world.WorldProvider;
+import org.terasology.world.block.Block;
+import org.terasology.world.block.BlockComponent;
+import org.terasology.world.block.BlockManager;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+@RegisterSystem(RegisterMode.AUTHORITY)
+public class UnificatorAction extends BaseComponentSystem {
+    private Side outputside;
```

Comment:
This variable is global for all Unificators in the world! If you will place more than one unificator block, it won't work. This variable should be store inside `UnificatorComponent` instead of being a system's property.